pkg/fake_pub_server/bin/fake_pub_server.dart

```diff
@@ -115,4 +130,58 @@ Future main(List<String> args) async {
     ],
     eagerError: true,
   );
+
+  await state.store();
+  await sigintSubscription.cancel();
+}
+
+class _LocalServerState {
+  final datastore = MemDatastore();
+  final storage = MemStorage();
+  Directory _dataDir;
+  File _datastoreFile;
+  File _storageFile;
+  Completer _storingCompleter;
+
+  _LocalServerState({String dataDir}) {
+    if (dataDir != null) {
+      _dataDir = Directory(dataDir);
+      _datastoreFile = File(p.join(dataDir, 'datastore.jsonl'));
+      _storageFile = File(p.join(dataDir, 'storage.jsonl'));
```

Comment:
Storing all buckets in a single file seems a bit extreme... but it's probably fine, since we won't store that much :D

wouldn't it be attractive to store it all in a single file, now that we are doing this anyways? That also makes it easier to send snapshots around.

We could just make the file:
```
"datastore-entries"
{...}
{...}
...
{...}
"storage-entries"
{...}
{...}
...
{...}
```

And use `takeWhile()` when reading it in `init()`.

Reply:
Good point, implementing it...